packages/attune/tests/it/main.rs

```diff
@@ -18,3 +31,333 @@ async fn migrations_applied(pool: sqlx::PgPool) {
+                .with_line_number(true)
+                .with_target(true)
+                .with_thread_ids(true)
+                .with_thread_names(true)
+                .with_writer(std::io::stderr)
+                .pretty(),
+        )
+        .with(tracing_subscriber::EnvFilter::from_default_env())
+        .init();
+
+    // Set up dependencies.
+    let sh = Shell::new().unwrap();
+    debug!(docker_host = ?std::env::var("DOCKER_HOST"), "connecting to Docker daemon");
+    let docker = Docker::connect_with_defaults().unwrap();
+    docker.ping().await.unwrap();
+
+    // HACK: We're using the `../../` to get to the workspace root because we
+    // know where this package is located, and the working directory is set to
+    // the package root in `cargo test`[^1].
+    //
+    // [^1]: https://github.com/rust-lang/cargo/issues/11852
+    let workspace_root = format!("{}/../..", env!("CARGO_MANIFEST_DIR"));
+
+    // Check that the CLI is built and executable.
+    const ATTUNE_CLI_PATH: &str = env!("CARGO_BIN_EXE_attune");
+    cmd!(sh, "{ATTUNE_CLI_PATH} --help").quiet().run().unwrap();
+    debug!(path = ?ATTUNE_CLI_PATH, "CLI binary accessible");
+
+    // // Run `docker compose up -d` to bring up the Docker services.
+    // cmd!(sh, "docker compose up -d").run().unwrap();
+
+    // // Monitor control plane for readiness.
+    // debug!("waiting for control plane");
+    // loop {
+    //     let status = docker
+    //         .inspect_container("attune-controlplane-1",
+    // None::<InspectContainerOptions>)         .await
+    //         .unwrap();
+    //     trace!(?status, "inspected control plane container status");
+    //     if status.state.unwrap().status.unwrap() ==
+    // ContainerStateStatusEnum::RUNNING {         break;
+    //     }
+    //     tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
+    // }
+
+    // Create a repository.
+    let repo = AttuneRepository::new(sh.clone(), ATTUNE_CLI_PATH.to_string());
+
+    // Add packages concurrently.
+    let gpg_key = GpgKey::new();
+    let key_id = &gpg_key.key_id;
+    let pubkey = cmd!(sh, "gpg --armor --export {key_id}").read().unwrap();
+
+    const DISTRIBUTIONS: [&str; 5] = ["stable", "unstable", "testing", "dev", "canary"];
+    const COMPONENTS: [&str; 8] = [
+        "main", "contrib", "non-free", "extra", "rolling", "cloud", "oss", "ee",
+    ];
+    const PACKAGES: [&str; 10] = [
+        "attune-test-package_2.0.0_linux_amd64.deb",
+        "attune-test-package_2.0.0_linux_arm64.deb",
+        "attune-test-package_3.0.5_linux_amd64.deb",
+        "attune-test-package_3.0.5_linux_arm64.deb",
+        "cod-test-package_3.0.5_linux_amd64.deb",
+        "cod-test-package_3.0.5_linux_arm64.deb",
+        "salmon-test-package_3.0.5_linux_amd64.deb",
+        "salmon-test-package_3.0.5_linux_arm64.deb",
+        "tuna-test-package_3.0.5_linux_amd64.deb",
+        "tuna-test-package_3.0.5_linux_arm64.deb",
+    ];
```

Comment:
[note] you can actually just let the compiler infer the count here with `_`
```suggestion
    const DISTRIBUTIONS: [&str; _] = ["stable", "unstable", "testing", "dev", "canary"];
    const COMPONENTS: [&str; _] = [
        "main", "contrib", "non-free", "extra", "rolling", "cloud", "oss", "ee",
    ];
    const PACKAGES: [&str; _] = [
        "attune-test-package_2.0.0_linux_amd64.deb",
        "attune-test-package_2.0.0_linux_arm64.deb",
        "attune-test-package_3.0.5_linux_amd64.deb",
        "attune-test-package_3.0.5_linux_arm64.deb",
        "cod-test-package_3.0.5_linux_amd64.deb",
        "cod-test-package_3.0.5_linux_arm64.deb",
        "salmon-test-package_3.0.5_linux_amd64.deb",
        "salmon-test-package_3.0.5_linux_arm64.deb",
        "tuna-test-package_3.0.5_linux_amd64.deb",
        "tuna-test-package_3.0.5_linux_arm64.deb",
    ];
```

Reply:
Doesn't work for me :(

```
error[E0121]: the placeholder `_` is not allowed within types on item signatures for constants
   --> packages/attune/tests/it/main.rs:143:33
    |
143 |     const DISTRIBUTIONS: [&str; _] = ["stable", "unstable", "testing", "dev", "canary"];
    |                                 ^ not allowed in type signatures
    |
help: replace this with a fully-specified type
    |
143 -     const DISTRIBUTIONS: [&str; _] = ["stable", "unstable", "testing", "dev", "canary"];
143 +     const DISTRIBUTIONS: [&str; 5] = ["stable", "unstable", "testing", "dev", "canary"];
    |

error[E0121]: the placeholder `_` is not allowed within types on item signatures for constants
   --> packages/attune/tests/it/main.rs:144:30
    |
144 |     const COMPONENTS: [&str; _] = [
    |                              ^ not allowed in type signatures
    |
help: replace this with a fully-specified type
    |
144 -     const COMPONENTS: [&str; _] = [
144 +     const COMPONENTS: [&str; 8] = [
    |

error[E0121]: the placeholder `_` is not allowed within types on item signatures for constants
   --> packages/attune/tests/it/main.rs:147:28
    |
147 |     const PACKAGES: [&str; _] = [
    |                            ^ not allowed in type signatures
    |
help: replace this with a fully-specified type
    |
147 -     const PACKAGES: [&str; _] = [
147 +     const PACKAGES: [&str; 10] = [
    |

For more information about this error, try `rustc --explain E0121`.
error: could not compile `attune` (test "it") due to 3 previous errors
```

Reply:
oh. hmm. guess i'm misremembering, could have sworn i did this before but maybe it was a crate's syntax. ah well.